### ISSUE TITLE
Changelog

### ISSUE THREAD

**Issue by Perkolator** [flathub/org.gimp.GIMP#132]
Could you please add a changelog to this project? I'd like to understand what's constantly updated even though GIMP itself hasn't been updated.

For example, after I updated the flatpak which bumped GIMP to 2.10.30 (btw, for some reason that flatpak update was available way before GIMP website announced the release), there has been few updates after that and I have no clue what is updated. I don't like to update anything if I don't know what's being updated. So please, could you add a changelog? Thank you.

**Comment by Jehan** [comment]
> Could you please add a changelog to this project?

Do you mean GIMP itself or the package only?

If you mean the package, I really don't see it necessary. This package barely has any updates a month (and I usually try to group them in order to avoid too many upgrades on user-side, though there were indeed quite a few this month, it seems that's why you are worried?) so for anyone who wants to know details, looking at the [public git log](https://github.com/flathub/org.gimp.GIMP/commits/master) should be enough and it can be parsed in just a few seconds.

Like here, you can see the last updates updated Exiv2, poppler, removed a useless dep of aalib, updated the runtime and released GIMP 2.10.30. So no we are not going to maintain a file to say exactly the same thing which is visible in public records already and can be read near-instantly. 😛

Now if you mean GIMP itself, then we already [publish web news](https://www.gimp.org/news/2021/12/21/gimp-2-10-30-released/), maintain a separate [NEWS](https://gitlab.gnome.org/GNOME/gimp/-/blob/05227c6889481e2ea973f81ccfa172150e057850/NEWS#L11-71) file with more details, and of course the [git log](https://gitlab.gnome.org/GNOME/gimp/-/commits/gimp-2-10) is visible as well.

> btw, for some reason that flatpak update was available way before GIMP website announced the release

The reason is that we are usually waiting for the Windows and macOS packages to be ready. Then when they are, we are waiting for all (or most at least) mirrors to be synced. Just the mirror thing takes up to 24 hours. Also we sleep (not said as a sarcasm at all: the last web news, nearly all mirrors were synced on the next evening, but there were still 2 or 3 with issues and I found it annoying enough not to want to release the news so I went to bed; I'm sure they got synced just a tiny bit after I went, but none was there to see it happen; hence the news happened to be published 1.5 days after actual source/flatpak release).

So yeah, the news is usually (always even, I'd say) announced on the website after GIMP is updated in flatpak, because I usually start the flatpak build immediately after I tag the version; hence it is available a few hours later, whereas the web news will be a day later at best (we try to improve this; it even used to be usually 3 or 4 days late not long ago). It is usually considered one of the advantages of this package.

> there has been few updates after that and I have no clue what is updated.

Actually flatpak has a feature to know what an update is about, through command line. Unfortunately Flathub doesn't display this info anywhere for end-user (as far as I am aware):

```sh
$ flatpak remote-info --user  flathub org.gimp.GIMP//stable
        ID: org.gimp.GIMP
       Ref: app/org.gimp.GIMP/x86_64/stable
      Arch: x86_64
    Branch: stable
Collection: org.flathub.Stable
  Download: 121.1 MB
 Installed: 349.4 MB
   Runtime: org.gnome.Platform/x86_64/41
       Sdk: org.gnome.Sdk/x86_64/41

    Commit: f8a41e43defd07b70f0b94b4619924440b5fe6292ae1f7982e193f3932e1d99b
    Parent: c07f87ee1334b99f141af4b33398a7ab50e8fe170c13f11727b73b17ce73f69f
   Subject: Update Exiv2 to 0.27.5 (74242e59)
      Date: 2021-12-26 00:30:22 +0000
```
Now say you want to know the release before that, then it's nearly the same command, but use the `parent` hash:
```sh
$ flatpak remote-info --user  flathub org.gimp.GIMP//stable --commit=c07f87ee1334b99f141af4b33398a7ab50e8fe170c13f11727b73b17ce73f69f
        ID: org.gimp.GIMP
       Ref: app/org.gimp.GIMP/x86_64/stable
      Arch: x86_64
    Branch: stable
Collection: org.flathub.Stable
  Download: 121.0 MB
 Installed: 349.1 MB
   Runtime: org.gnome.Platform/x86_64/41
       Sdk: org.gnome.Sdk/x86_64/41

    Commit: c07f87ee1334b99f141af4b33398a7ab50e8fe170c13f11727b73b17ce73f69f
    Parent: c353ff42810b2b8aca6f76928577761143edefea82b2f50a6ef6be45e8f308ac
   Subject: Issue #129 - remove slang from aalib (ef6a6727)
      Date: 2021-12-23 06:37:31 +0000
```

Yet this is not perfect, as you'll notice it just uses the message of the last commit (and there were 3 commits in the last package publication). But once again, the [git log](https://github.com/flathub/org.gimp.GIMP/commits/master) makes the best Changelog.

Moreover I'd like to note that this package is an official package maintained by the GIMP team. So if it's a trust thing (which I completely understand, and I commend you for this), you can have as much trust in this package as in the main source anyway. I am maintainer on both GIMP source code and this package and I double-check every commit going in there (as in the source code), same as I assume that @hfiguiere double-checks mine here (since he regularly comments when things are not so good, which improves the package quality) and others in GIMP source code. It's kind of a live peer review common to Free Software. Now I'd say it's up to you to decide if you can trust us all checking each other's code. 😃

> So please, could you add a changelog?

In GIMP source, we have one/many already, in the package, as said above, the [gitlog](https://github.com/flathub/org.gimp.GIMP/commits/master) should be enough and there is not enough volume to deserve a redundant file. Yet I hope that the proposed alternative(s) is/are enough.

@hfiguiere You may have noticed that I tended not to merge patches too quickly even when sometimes I was saying they were good. The reason was because I was trying to combine them in order to avoid multiple updates in a row which can be annoying to people (and even scary, as we can see here). Maybe we could make up some official rule for this? Unless something is an emergency (like big discovered vulnerability or something), when we plan a series of change, let's try to push them at once? For instance, between the 21 to 26, you pushed 3 package updates (wonderful, by the way! 👍). Maybe the next time, once you validate the first, continue to build on top of it, and push all of them at once? 🙂

**Comment by Perkolator** [comment]
> This package barely has any updates a month (and I usually try to group them in order to avoid too many upgrades on user-side, though there were indeed quite a few this month, it seems that's why you are worried?) so for anyone who wants to know details, looking at the [public git log](https://github.com/flathub/org.gimp.GIMP/commits/master) should be enough and it can be parsed in just a few seconds.

Thanks for that link. I've never seen a page like that in github before, never knew where I could click to reach that kind of page. (and I even have published one repo myself.. very simple repo though..) I guess it's hard to find for even average users.. or I'm very stupid, which is hard to know because only smart people know if they are stupid. ;)

> Actually flatpak has a feature to know what an update is about, through command line. Unfortunately Flathub doesn't display this info anywhere for end-user (as far as I am aware):
> 
> ```shell
> $ flatpak remote-info --user  flathub org.gimp.GIMP//stable
> ```

Thanks for the hint, always appreciate learning more. Though, as you said, it only shows last commit message. Also, I had to remove the `--user` option, it spat out this: "error: Remote "flathub" not found in the user installation".

Thanks a lot for the response! I guess I'm a dying breed that cares about what is actually updated in software world. I see more and more automatic silent updates everywhere and people seem to be fine with them. I never auto-update anything, not even in my phone (though some vendors make it very hard to know what has been updated.. \*cough\* google...), I even glimpse through kernel updates for my linux installation (in case there's something related to some little problems/shortcomings I've had encountered while playing in the linux world).
